lib/puppet/functions/cache_data.rb

```diff
@@ -0,0 +1,40 @@
+require 'fileutils'
+require 'yaml'
+require 'etc'
```

Comment:
where is that coming from? I assume https://rubygems.org/gems/etc ?

Reply:
It was already there. It's https://ruby-doc.org/stdlib-2.5.1/libdoc/etc/rdoc/Etc.html

Reply:
ah nice.